python/docs/basic.rst

```diff
@@ -0,0 +1,85 @@
+Basic Usage
+===========
+
+annotate
+--------
+
+The ``annotate()`` function annotates a code range, i.e., one or more statements.
+Each annotation may have a message and a color associated with it.
+This makes it easy to distinguish annotated ranges when visualizing them.
+``annotate`` can be used in two ways:
+
+As a decorator:
+::
+
+   @nvtx.annotate(message="my_message", color="blue")
+   def my_func():
+       pass
+
+
+As a context manager:
+::
+
+   with nvtx.annotate(message="my_message", color="green"):
+       pass
+
+
+When used as a decorator, the ``message`` argument defaults to the
+name of the function being decorated:
+::
+
+   @nvtx.annotate()  # message defaults to "my_func"
+   def my_func():
+       pass
+
+
+mark
+----
+
+The ``mark()`` function marks an instantaneous event in the execution of a program.
+For example, you may want to mark when an exceptional event occurs:
+::
+
+   try:
+       something()
+   except SomeError():
+       nvtx.mark(message="some error occurred", color="red")
+       # ... do something else ...
+
+
+Domains
+-------
+
+In addition to a message and a color,
```

Comment:
There should be more documentation that Domains are really intended to be 1 per library. Lots of people tend to abuse Domains when they should really be using Categories.

Reply:
Speaking of which, did Categories not make v1?

Reply:
This is `v1` so it's not too late :)

Reply:
Added and documented categories

Reply:
Also a note about when to use domains